### PR TITLE
Upgrading to .net 6.0

### PR DESCRIPTION
**Pull request by eladiw**
The following changes are proposed:

* Upgrading the .net version to latest (6.0)
* Had to remove the use of WebClient which is obsolete since .net 6

### CODE REVIEW

File: K2Bridge.Tests.End2End/KustoTestBase.cs [hunk 1]
```diff
@@ -118,12 +118,29 @@ public static async Task Init()
...
 
             kusto = new KustoConnectionStringBuilder(kustoUri, kustoDatabase)
-               .WithAadApplicationTokenAuthentication(token);
+                .WithAadApplicationTokenAuthentication(token);
 
             if (!File.Exists("flights.json.gz"))
             {
-                using var wc = new WebClient();
-                wc.DownloadFile("https://raw.githubusercontent.com/elastic/kibana/v6.8.5/src/server/sample_data/data_sets/flights/flights.json.gz", "flights.json.gz");
+                using var hc = new HttpClient();
```
Comment (AsafMah) [inline K2Bridge.Tests.End2End/KustoTestBase.cs:125]: Move this into a function?

File: K2Bridge.Tests.End2End/KustoTestBase.cs [hunk 2]
```diff
@@ -118,12 +118,29 @@ public static async Task Init()
...
+                var response =
+                    await hc.GetAsync(new Uri(
+                        "https://raw.githubusercontent.com/elastic/kibana/v6.8.5/src/server/sample_data/data_sets/flights/flights.json.gz"));
+                if (response.IsSuccessStatusCode)
+                {
+                    using (var stream = await response.Content.ReadAsStreamAsync())
+                    {
+                        var fi = new FileInfo("flights.json.gz");
+                        using (var fs = fi.OpenWrite())
```
Comment (AsafMah) [inline K2Bridge.Tests.End2End/KustoTestBase.cs:134]: If you're using the "using declaration" above, you can do it here to:

```c#
                    using var stream = await response.Content.ReadAsStreamAsync();
                    var fi = new FileInfo("flights.json.gz");
                    using var fs = fi.OpenWrite();
                    await stream.CopyToAsync(fs);
```
You can upgrade it further by using "await using", which will dispose asynchronously:
```c#
                    await using var stream = await response.Content.ReadAsStreamAsync();
                    var fi = new FileInfo("flights.json.gz");
                    await using var fs = fi.OpenWrite();
                    await stream.CopyToAsync(fs);
```

Also, is that really the minimal amount of code to achieve the same functionallity as WebClient?

File: K2Bridge.Tests.End2End/TestElasticClient.cs [hunk 3]
```diff
@@ -120,7 +120,7 @@ public async Task<JToken> MSearch(string indexName, string jsonQueryFile, bool v
 
             using var request = new HttpRequestMessage(HttpMethod.Post, "_msearch");
             var payload = new StringBuilder();
-            payload.AppendLine($"{{\"index\":\"{indexName}\"}}");
+            payload.AppendLine(FormattableString.Invariant($"{{\"index\":\"{indexName}\"}}"));
```
Comment (AsafMah) [inline K2Bridge.Tests.End2End/TestElasticClient.cs:123]: Why is it needed? Why only here?

Comment (eladiw) [reply K2Bridge.Tests.End2End/TestElasticClient.cs:123]: This is the only place the compiler issued a warning

File: K2Bridge.Tests.End2End/KustoTestBase.cs (hunk 2 above)
Comment (eladiw) [reply K2Bridge.Tests.End2End/KustoTestBase.cs]: about your last comment, the answer is yes.

File: K2Bridge.Tests.End2End/KustoTestBase.cs [hunk 4]
```diff
@@ -211,5 +211,29 @@ private static async Task<TestElasticClient> CreateKustoClient(string prefix)
...
+        /// <summary>
+        /// Download a remote file.
+        /// </summary>
+        /// <param name="url">the URL of the remote file</param>
+        /// <param name="fileName">The file to be created</param>
+        /// <exception cref="FileNotFoundException"></exception>
+        private static async Task DownloadFile(string url, string fileName)
+        {
+            using var hc = new HttpClient();
+            var response =
```
Comment (AsafMah) [inline K2Bridge.Tests.End2End/KustoTestBase.cs:224]: nit: no need for a line break

**Review by AsafMah — APPROVED** [review]

**Pull request merged**